### PR TITLE
Added tabs and tab panel paragraphs

### PR DESCRIPTION
**Pull request by gosia-mlynarczyk**
Relates to: [Add Tabs paragraph](https://github.com/localgovdrupal/localgov_campaigns/issues/42)

I've added 2 new paragraphs (Tabs and its child paragraph Tab panel).
I've also made some changes to the Accordion paragraph's script, as the Tabs component can be displayed as an Accordion on smaller devices (that's how it currently works on the Bracknell site).

### CODE REVIEW

**Comment by finnlewis** [comment]
Hey @gosia-mlynarczyk 
The tests pass, but it looks like there are conflicting files on this pull reuqest. Could you review?

**Comment by gosia-mlynarczyk** [comment]
Thanks @finnlewis , that's done now.

**Comment by finnlewis** [comment]
I've tested this locally and the tabs paragraph works well. 

All the tests pass and no merge conflicts. 

The keyboard navigation is neat, so I'm guessing this is functionality that has been tested elsehwere for accessibility @gosia-mlynarczyk , is that right? 

Perhaps one quick review of the javascript from another developer: 

@andybroomfield @stephen-cox @Adnan-cds any of you want to just confirm you're happy with the JS?

https://github.com/localgovdrupal/localgov_campaigns/pull/59/files?file-filters%5B%5D=.js#diff-5dd7a8947cbc1cc5b4521aea4295220f
https://github.com/localgovdrupal/localgov_campaigns/pull/59/files?file-filters%5B%5D=.js#diff-dcc1af500ab29eeb4a2f7f47541480d7

File: modules/localgov_campaigns_paragraphs/js/localgov-tabs.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+      }
+
+      // Set new active panel.
+      newActivePanel[0].classList.add(activePanelClass);
+      if (oldActiveTab) {
+        newActivePanel[0].focus();
+      }
+    }
+  };
+})();
```
Comment (Adnan-cds) [inline modules/localgov_campaigns_paragraphs/js/localgov-tabs.js:227]: Usually, this is where you would pass in the "Drupal" object and then accept it as a function parameter above.  Drupal core has some good [examples](https://git.drupalcode.org/project/drupal/-/blob/9.0.x/core/misc/checkbox.js).

File: modules/localgov_campaigns_paragraphs/js/localgov-tabs.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+      var create = function create() {
+        if (!tabs.classList.contains(tabsInitialisedClass)) {
+          var _ret = function () {
+            var tabPanelsNumber = tabPanels.length;
+
+            // Only initialise tabs if there are at least 2 panels.
+            if (tabPanelsNumber < 2) {
+              return {
+                v: void 0
```
Comment (Adnan-cds) [inline modules/localgov_campaigns_paragraphs/js/localgov-tabs.js:45]: MDN has something interesting to [say about the use of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void#Description) ```void 0```.

File: modules/localgov_campaigns_paragraphs/js/localgov-tabs.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
+/**
+ * @file
+ * Tabs behaviour.
+ */
+
+(function () {
+  Drupal.behaviors.tabs = {
```
Comment (Adnan-cds) [inline modules/localgov_campaigns_paragraphs/js/localgov-tabs.js:7]: ```Drupal.behaviors.localgovTabs``` would be even better.

File: modules/localgov_campaigns_paragraphs/js/localgov-accordion.js [hunk 4]
```diff
@@ -32,63 +32,111 @@
...
+          destroy();
+        }
+      };
 
-        // Add init class.
-        accordion.classList.add(initClass);
+      // Trigger create/destroy functions at different screen widths
+      // based on the value of data-accordion-tabs-switch attribute.
+      if (window.matchMedia) {
+        var mq = window.matchMedia(`(max-width: ${breakpoint})`);
```
Comment (Adnan-cds) [inline modules/localgov_campaigns_paragraphs/js/localgov-accordion.js:135]: ES6 Templates literals [don't work in IE11](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Browser_compatibility).  We, at least in Croydon, have to support IE11.

I can see two options here:
- Stop supporting IE11 :)
- Follow the [lead of Drupal core](https://www.drupal.org/node/2815083) and provide two copies of this file - localgov-accordion.es6.js and localgov-accordion.js.  The later produced by Babel.

@gosia-mlynarczyk @finnlewis @cjstevens78

**Comment by Adnan-cds** [comment]
> any of you want to just confirm you're happy with the JS?

Hi Finn, Gosia,
JS code looks mostly okay.  I have added several comments.  I must say I haven't tested the code in a browser.

FAO @finnlewis @gosia-mlynarczyk

File: modules/localgov_campaigns_paragraphs/js/localgov-tabs.js (hunk 1 above)
Comment (gosia-mlynarczyk) [reply modules/localgov_campaigns_paragraphs/js/localgov-tabs.js]: Added

File: modules/localgov_campaigns_paragraphs/js/localgov-tabs.js (hunk 3 above)
Comment (gosia-mlynarczyk) [reply modules/localgov_campaigns_paragraphs/js/localgov-tabs.js]: You're right, updated

File: modules/localgov_campaigns_paragraphs/js/localgov-accordion.js (hunk 4 above)
Comment (gosia-mlynarczyk) [reply modules/localgov_campaigns_paragraphs/js/localgov-accordion.js]: I vote for option 1! 😉 

Ok, I've added source files (.es6.js) for both the accordion and tabs.

File: modules/localgov_campaigns_paragraphs/js/localgov-tabs.js (hunk 2 above)
Comment (gosia-mlynarczyk) [reply modules/localgov_campaigns_paragraphs/js/localgov-tabs.js:51]: This is actually how ES6 code is compiled. 
So this:
```
 if (tabPanelsNumber < 2) {
   return;
 }
```
becomes this:
```
if (tabPanelsNumber < 2) {
  return {
    v: void 0
  };
}
```

Comment (Adnan-cds) [reply modules/localgov_campaigns_paragraphs/js/localgov-tabs.js:51]: Thanks for the clarification Gosia :)

File: modules/localgov_campaigns_paragraphs/js/localgov-tabs.js (hunk 1 above)
Comment (Adnan-cds) [reply modules/localgov_campaigns_paragraphs/js/localgov-tabs.js]: Thanks :)

File: modules/localgov_campaigns_paragraphs/js/localgov-accordion.js (hunk 4 above)
Comment (Adnan-cds) [reply modules/localgov_campaigns_paragraphs/js/localgov-accordion.js]: +10 for adding the ES6 source file.

**Review by Adnan-cds — APPROVED** [review]
Hi Finn, Gosia,
I am happy with the Javascript :)

FAO @finnlewis @gosia-mlynarczyk

**Comment by gosia-mlynarczyk** [comment]
Thanks @Adnan-cds !

**Pull request merged**